thread.c

```diff
@@ -2223,10 +2223,10 @@ handle_interrupt_arg_check_i(VALUE key, VALUE val, VALUE args)
  * standard library Timeout, from lib/timeout.rb
  *
  *   require 'timeout'
- *   Thread.handle_interrupt(Timeout::Error => :never) {
- *     timeout(10){
+ *   Thread.handle_interrupt(Timeout::ExitException => :never) {
+ *     Timeout.timeout(10){
  *       # Timeout::Error doesn't occur here
```

Comment:
The specific error message on L2228, should it _also_ be ExitException?

Reply:
It's a good question - I left it that way only because it still seems you still should _catch_ Timeout::Error, not ExitException.

@jeremyevans I think you made that ExitException change - what do you think is clearer here?

Reply:
Mmh, I don't understand thsi example well, why is it even using `handle_interrupt` here?
It looks useless to me.
Or maybe the example needs to be more complete and actually include an `ensure` to illustrate what it wants to do.
In the current state I think it would be better to remove this confusing example.

Reply:
I agree with @eregon, we should remove the entire "Guarding from Timeout::Error" section.

Reply:
Ok, I'll remove it with another commit